### PR TITLE
fix thread count initialization

### PR DESCRIPTION
**Pull request by pixelflinger**
if hw thread count was 1, we'd end-up with 32 threads

### CODE REVIEW

File: libs/utils/src/JobSystem.cpp [hunk 1]
```diff
@@ -105,30 +105,34 @@ void JobSystem::setThreadAffinityById(size_t id) noexcept {
...
             // For now we avoid using HT, this simplifies profiling.
             // TODO: figure-out what to do with Hyper-threading
-            threadCount = hwThreads / 2 - 1;
-        } else {
-            threadCount = hwThreads - 1;
+            // since we assumed HT, always round-up to an even number of cores (to play it safe)
+            hwThreads = (hwThreads + 1) / 2;
         }
+        // make sure we have at least one h/w thread (could be an assert instead)
+        hwThreads = std::max(0, hwThreads);
```
Comment (romainguy) [inline libs/utils/src/JobSystem.cpp]: When can hwThread ever be negative? Did you mean max(1,...)? That's what the comment says.

File: libs/utils/src/JobSystem.cpp [hunk 2]
```diff
@@ -105,30 +105,34 @@ void JobSystem::setThreadAffinityById(size_t id) noexcept {
...
+        // one of the thread will be the user thread
+        threadPoolCount = hwThreads - 1;
```
Comment (romainguy) [inline libs/utils/src/JobSystem.cpp]: If we have only 1 hw thread this will be 0. Expected?

File: libs/utils/src/JobSystem.cpp [hunk 3]
```diff
@@ -105,30 +105,34 @@ void JobSystem::setThreadAffinityById(size_t id) noexcept {
...
     }
-    threadCount = std::min(size_t(UTILS_HAS_THREADING ? 32 : 0), threadCount);
+    threadPoolCount = std::min(UTILS_HAS_THREADING ? 32 : 0, threadPoolCount);
```
Comment (romainguy) [inline libs/utils/src/JobSystem.cpp]: Btw why only 32 threads if the CPU has 48 cores for instance?

File: libs/utils/src/JobSystem.cpp (hunk 1 above)
Comment (pixelflinger) [reply libs/utils/src/JobSystem.cpp]: it should never happen, that's why "it could be an assert".

File: libs/utils/src/JobSystem.cpp (hunk 2 above)
Comment (pixelflinger) [reply libs/utils/src/JobSystem.cpp]: yes -- it means we won't process any jobs in the thread-pool. they'll all be processed in the user thread when they call wait().

File: libs/utils/src/JobSystem.cpp (hunk 3 above)
Comment (pixelflinger) [reply libs/utils/src/JobSystem.cpp]: I don't remember why this was there

**Pull request merged**